os/fs/driver/mtd/smart.c

```diff
@@ -521,6 +524,27 @@ static const struct block_operations g_bops = {
  ****************************************************************************/
 
 #ifdef CONFIG_DEBUG_FS
+static int smart_dumpall(FAR struct smart_struct_s *dev, FAR char* buffer)
+{
+	int ret = OK;
+	uint32_t sector = 0;
+
+	if (dump_flag) {
```

Comment:
If you apply my last comment regarding sector size, you can remove dump_flag here.

Reply:
Why resolved this without any comment?
dump_flag is not necessary if we know about sectorsize, and we can get it through the procfs.
Anyway I think dump flag is not unsafe for multi thread... make sure that this should work with another module, not only with dump tool.

Reply:
Done by mistake, sorry.
Updated now.